### PR TITLE
Ghost MSP support

### PR DESCRIPTION
**Pull request by daleckystepan**
Add MSP support for Ghost

Need new Ghost FW v1.0.6.0+

BF: https://github.com/betaflight/betaflight/pull/11242
ETX: https://github.com/EdgeTX/edgetx/pull/1377

### CODE REVIEW

**Review by haslinghuis — APPROVED** [review]

**Comment by TheIsotopes** [comment]
@daleckystepan The only thing i missing is the public beta of Ghost FW 1.0.6.0 and also the support for OpenTX, then I could test this too.

**Comment by daleckystepan** [comment]
@TheIsotopes It is not ready yet for testing. I will let you know once it will be. Intention for this PR is to create place where we can discuss it etc. You are really fast :D

**Comment by klutvott123** [comment]
@daleckystepan Good job on this and the rest of the PRs to go with it 👍 
I'm moving this to the 1.7 milestone since the firmware side of it is for 4.4.

File: src/SCRIPTS/BF/MSP/ghost.lua [hunk 1]
```diff
@@ -0,0 +1,28 @@
+-- GHOST Frame Types
+local GHST_FRAMETYPE_MSP_REQ           = 0x21
+local GHST_FRAMETYPE_MSP_WRITE         = 0x22
+local GHST_FRAMETYPE_MSP_RESP          = 0x28
+
+ghstMspType = 0
```
Comment (klutvott123) [inline src/SCRIPTS/BF/MSP/ghost.lua:6]: This can be ```local ghstMspType = 0```

File: src/SCRIPTS/BF/MSP/ghost.lua [hunk 2]
```diff
@@ -0,0 +1,28 @@
+-- GHOST Frame Types
+local GHST_FRAMETYPE_MSP_REQ           = 0x21
+local GHST_FRAMETYPE_MSP_WRITE         = 0x22
+local GHST_FRAMETYPE_MSP_RESP          = 0x28
```
Comment (klutvott123) [inline src/SCRIPTS/BF/MSP/ghost.lua:4]: Can we have fewer spaces here? 
I see that were calling it both "ghst" and "ghost". Which one is the "correct" one?

File: src/SCRIPTS/BF/protocols.lua [hunk 3]
```diff
@@ -28,6 +38,8 @@ local function getProtocol()
         return supportedProtocols.smartPort
     elseif supportedProtocols.crsf.push() ~= nil then
         return supportedProtocols.crsf
+    elseif supportedProtocols.ghost.push() ~= nil then
+        return supportedProtocols.ghost
```
Comment (haslinghuis) [inline src/SCRIPTS/BF/protocols.lua:42]:
```suggestion
        return supportedProtocols.ghst
```

File: src/SCRIPTS/BF/protocols.lua [hunk 4]
```diff
@@ -20,6 +20,16 @@ local supportedProtocols =
         saveMaxRetries  = 2,
         saveTimeout     = 150,
         cms             = {},
+    },
+    ghost =
```
Comment (haslinghuis) [inline src/SCRIPTS/BF/protocols.lua:24]:
```suggestion
    ghst =
```

File: src/SCRIPTS/BF/MSP/ghost.lua (hunk 2 above)
Comment (haslinghuis) [reply src/SCRIPTS/BF/MSP/ghost.lua:4]: I prefer `ghst` as it goes nice with `csrf`

File: src/SCRIPTS/BF/MSP/ghost.lua (hunk 1 above)
Comment (daleckystepan) [reply src/SCRIPTS/BF/MSP/ghost.lua]: Fixed

File: src/SCRIPTS/BF/MSP/ghost.lua (hunk 2 above)
Comment (daleckystepan) [reply src/SCRIPTS/BF/MSP/ghost.lua]: Fixed

**Review by klutvott123 — APPROVED** [review]

**Comment by carbendi** [comment]
Is this working now? 
Getting stuck at "downloading board info" after compiling and running the betaflight lua script. 
Jumper T16 
Edgetx 2.7.1
BF 4.4 latest
Ghost 1.0.6.5

**Comment by haslinghuis** [comment]
Try latest nightly viewer: https://github.com/betaflight/blackbox-log-viewer-nightlies/releases/tag/v20220921.21

**Pull request merged**